peanut_gb.h

```diff
@@ -882,14 +994,26 @@ void __gb_write(struct gb_s *gb, uint_fast16_t addr, uint8_t val)
 		if(gb->mbc == 1)
 		{
 			gb->cart_ram_bank = (val & 3);
+#if PEANUT_FULL_GBC_SUPPORT
+			gb->cart_ram_bank_offset = 0xA000 - (gb->cart_ram_bank << 13);
+#endif
 			gb->selected_rom_bank = ((val & 3) << 5) | (gb->selected_rom_bank & 0x1F);
 			gb->selected_rom_bank = gb->selected_rom_bank & gb->num_rom_banks_mask;
 		}
 		else if(gb->mbc == 3)
+		{
 			gb->cart_ram_bank = val;
+#if PEANUT_FULL_GBC_SUPPORT
+			gb->cart_ram_bank_offset = 0xA000 - ((gb->cart_ram_bank & 3) << 13);
+#endif
+		}
 		else if(gb->mbc == 5)
+		{
 			gb->cart_ram_bank = (val & 0x0F);
-
+#if PEANUT_FULL_GBC_SUPPORT
+			gb->cart_ram_bank_offset = 0xA000 - (gb->cart_ram_bank << 13);
+#endif
+		}
 		return;
```

Comment:
Are you sure that the memory bank controllers of the cartridge changes behaviour with CGB mode?